### PR TITLE
Revert "MM-45956: Optimize FileInfo stats query (#22603)"

### PR DESCRIPTION
**Pull request by lieut-data**
#### Summary
This reverts commit 56b18ca7bfcf1bc513fe8bbf45dc1c0b010bc4de.

During the cloud release, an unexpected issue occurred updating a small number of installations related to this migration. While work continues trying to understand the root cause, this PR gives the option to simply revert and unblock the release.

It reverts cleanly, and requires no clean up since the schema changes are backwards compatible. If this PR is merged and we later restore the improvement, a new migration will be necessary to ensure that Cloud installations that ran prior to this PR update once again.

#### Ticket Link
Relates-to: https://mattermost.atlassian.net/browse/MM-45956

#### Screenshots
```release-note
NONE
```

### CODE REVIEW

File: server/channels/db/migrations/migrations.list
Comment (agarciamontoro) [inline server/channels/db/migrations/migrations.list:222]: Leftovers from some merge conflicts.

Comment (lieut-data) [reply server/channels/db/migrations/migrations.list]: Ooh, good catch! I'm actually curious: what do we use this file for?

Comment (agarciamontoro) [reply server/channels/db/migrations/migrations.list]: That's a good question.

**Review by agarciamontoro — APPROVED** [review]
Sad to see the optimization go. (I'm optimistic we'll get it back soon, though)

**Comment by streamer45** [comment]
@lieut-data To confirm, this was never released on self-hosted?

**Review by streamer45 — APPROVED** [review]

**Comment by lieut-data** [comment]
@streamer45, correct: this change was first introduced just after the monorepo, which would ship for the first time in v7.11 next month.

**Comment by lieut-data** [comment]
Heads up I've rebased this against `cloud`. Now that we've root caused the issue, it's no longer necessary to revert this in `master`. We will only merge this if @spirosoik and @stylianosrigas think it's necessary to derisk the Cloud release, but would expect to resolve the issues in Cloud before the subsequent release.

**Review by agarciamontoro — APPROVED** [review]
Changes look good. I'll defer to Spiros and Stelios whether we want to merge this or not.

**Review by spirosoik — APPROVED** [review]
Thanks again

**Comment by agarciamontoro** [comment]
My force-pushed commit was created with:
- Hard-reset `revert-fileinfo-perf-improvements` to current `cloud`
- Reverted commit https://github.com/mattermost/mattermost-server/commit/56b18ca7bfcf1bc513fe8bbf45dc1c0b010bc4de

It was merged with no conflicts, but I'd like another pair of eyes on it.

**Comment by lieut-data** [comment]
This proved to be unnecessary, as we resolved the issue elsewhere: https://github.com/mattermost/mattermost-operator/pull/350

**Pull request closed without merging**
